tasks/libs/common/git.py

```diff
@@ -45,6 +45,40 @@ def get_staged_files(ctx, commit="HEAD", include_deleted_files=False) -> Iterabl
                 yield file
 
 
+def get_file_modifications(
+    ctx, base_branch=DEFAULT_BRANCH, added=False, modified=False, removed=False, only_names=False
+) -> list[tuple[str, str]]:
+    """
+    Get file status changes for the current branch compared to the base branch.
+
+    If no filter is provided, will return all the files.
+
+    - added: Include added files
+    - modified: Include modified files
+    - removed: Include removed files
+    - only_names: Return only the file names without the status
+    - Returns a list of (status, filename)
+    """
+
+    last_main_commit = ctx.run(f"git merge-base HEAD origin/{base_branch}", hide=True).stdout.strip()
+
+    modifications = [
+        line.split() for line in ctx.run(f"git diff --name-status {last_main_commit}", hide=True).stdout.splitlines()
+    ]
```

Comment:
What about using [this](https://github.com/DataDog/datadog-agent/blob/main/tasks/libs/common/git.py#L48)?

Reply:
The problem with this function is that it doesn't return the status nor the renamed files. I will update my function such that `get_modified_files` will rely on it